lib/spack/spack/spec.py

```diff
@@ -3769,6 +3769,17 @@ def write(s, c=None):
         result = out.getvalue()
         return result
 
+    @property
+    def uarch_flags(self):
+        """Specific target optimization flags."""
+        if not self.concrete:
+            return None
+
+        return self.target.optimization_flags(
+            self.compiler.name,
+            self.compiler.version,
+        )
```

Comment:
Using this directly I think we need to manage the `UnsupportedMicroarchitecture` exception for edge cases like mixed compiler toolchain. Otherwise you can use:
```python
self.arch.target.optimization_flags()
```
which takes care of these edge cases before delegating to the call above (but will always emit a warning).